refinery/core/api.py

```diff
@@ -282,6 +282,12 @@ def obj_get(self, bundle, **kwargs):
         if not user.is_authenticated() and res and not res.is_public():
             return HttpUnauthorized()
 
+        if not res.is_valid():
+            raise NotFound(
+                "DataSet with UUID: {} is invalid, and most likely is "
```

Comment:
The `super` calls (or regular calls we were making prior) being made to`obj_get` with a uuid of an invalid DataSet are followed directly by a call to `dehydrate()`. This was the root cause of the `AttributeErrors` we were seeing originally. 

The logic to handle this has to be in `DataSetResource.dehydrate()`, `DataSetResource.obj_get()`, or `SharableResourceAPIInterface.obj_get()` AFAIK.

I introduced logic to handle for these invalid DataSets which I provided in the `deyhdrate()` method originally like so:
``` python
class DataSetResource(ModelResource, SharableResourceAPIInterface):
...
    def dehydrate(self, bundle):
        if not bundle.obj.is_valid():
            # Dataset has not been associated with its InvestigationLink so
            # we are not able to provide a fully created bundle. Returning
            # the bundle in this state will properly exclude it from the api
            # response
            return bundle
    ...
```
Then I was asked to move it out of `dehydrate()`, so I moved it into `obj_get()` like so:
``` python
class DataSetResource(ModelResource, SharableResourceAPIInterface):
...
    def obj_get(self, bundle, **kwargs):
            try:
                dataset = DataSet.objects.get(uuid=kwargs["uuid"])
            except (DataSet.DoesNotExist, DataSet.MultipleObjectsReturned) as e:
                logger.error("Couldn't properly fetch DataSet with UUID: %s %s",
                             kwargs["uuid"], e)
                raise NotFound(e)

            if not dataset.is_valid():
                raise NotFound(
                    "DataSet with UUID: {} is invalid, and most likely is "
                    "still being created".format(dataset.uuid)
                )
            return SharableResourceAPIInterface.obj_get(self, bundle, **kwargs)
```

Then, being asked to remove that code block forced me to place the logic in the parent class that your are seeing here.